### DISCUSSION TITLE
Is it possible to specify different lr for different layer or fix some layer？

### DISCUSSION THREAD

**Discussion by funan-jhc-lee** [deepmodeling/deepmd-kit#325] [Q&A] [answered]
Hello！As stated in the title, is it possible to specify different **lr** for different layer or fix some layer, when retraining a model？I‘ve read the document, but It seems not possible.

**Comment by amcadmus** [comment, marked as answer]
Hello @funan-jhc-lee 

DeepMD-kit does not support different lr in different layers.  

You can fix the parameters in the embedding net by setting `trainable` to `False`, or fix the parameters of the hidden and output layers of the fitting net by a list of booleans provided to `trainable`. 

Please refer to https://github.com/deepmodeling/deepmd-kit/blob/master/doc/train-input-auto.rst 
`model/descriptor[se_a]/trainable` and
`model/fitting_net[ener]/trainable`
for more information

**Reply by funan-jhc-lee** [reply]
Thanks a lot! I think It's a quite useful feature that allows users specify the lr for diffrent layers or add more neurons(more layers) when retraining a model. Is it possible that this feature will be added in a future update? Thanks again!
